diff -crB marisa-trie-original/include/marisa/exception.h marisa-trie/include/marisa/exception.h
*** marisa-trie-original/include/marisa/exception.h	2025-01-11 00:50:21.294560568 +0000
--- marisa-trie/include/marisa/exception.h	2025-01-11 00:51:30.300379478 +0000
***************
*** 19,25 ****
    Exception(const Exception &ex)
        : std::exception(), filename_(ex.filename_), line_(ex.line_),
          error_code_(ex.error_code_), error_message_(ex.error_message_) {}
!   virtual ~Exception() throw() {}
  
    Exception &operator=(const Exception &rhs) {
      filename_ = rhs.filename_;
--- 19,25 ----
    Exception(const Exception &ex)
        : std::exception(), filename_(ex.filename_), line_(ex.line_),
          error_code_(ex.error_code_), error_message_(ex.error_message_) {}
!   virtual ~Exception() noexcept;
  
    Exception &operator=(const Exception &rhs) {
      filename_ = rhs.filename_;
diff -crB marisa-trie-original/lib/marisa/agent.cc marisa-trie/lib/marisa/agent.cc
*** marisa-trie-original/lib/marisa/agent.cc	2025-01-11 00:50:21.304214154 +0000
--- marisa-trie/lib/marisa/agent.cc	2025-01-11 00:51:30.302165729 +0000
***************
*** 5,10 ****
--- 5,12 ----
  
  namespace marisa {
  
+ Exception::~Exception() noexcept {}
+ 
  Agent::Agent() : query_(), key_(), state_() {} 
  
  Agent::~Agent() {}
